### PR TITLE
create fallback turn server option

### LINKED ISSUE #2329: allow to use fallbackcall assist server (turn.matrix.org)
it seems that element android does not allow to use the fallback call assist server (turn.matrix.org), while riot android did, and element web does.

this option is useful for people running their own homeserver without a turn server. ideally, each homeserver should have its turn server, but it is not always possible to run one. it can be because of network traffic or computing power limitations, or because of network setup constraints. for example, when the homeserver is running in a home network behind nat and has a different ip address inside and outside the home network, running a turn server on it does not seem to work (or if it is, its configuration seems pretty challenging).

will you please add this option?

### LINKED ISSUE #3063: ability to switch on using turn server turn.matrix.org from Android device
I urgently needed to set up a video call with someone. I was using element on linux on a PC and he was using element on an android tablet on WiFi installed from fdroid. We could text message but when you tried to make a video or audio call it refused to connect. 

On my linux element I switched on connect to matrix turn server:

> 
> Einstellungen > Sprach- & Videoanruf > 
> „Auf den Server turn.matrix.org zurückgreifen, falls deine Heimserver keine Anruf-Assistenz anbietet (deine IP-Adresse wird während eines Anrufs geteilt)“ 

Settings - video and voice calls > use turn.matrix.org if the homeserver offers no call assist (your ip address will be shared during the call)

That was not enough. 

I asked him to do the same and he said it was not there. 

Now I installed element myself on an Android device and note that indeed it definitely is NOT there!!

Could you please add it? It is very important. If it is hiding somewhere please tell me where it is..

### PR DESCRIPTION
**Pull request by CicadaCinema**
Fixes #2329 
Fixes #3063

# PR Moved from https://github.com/vector-im/element-android/pull/4391

## Screenshot
You need to enable the fallback option in `Settings > Voice & Video` to use this feature.

### CODE REVIEW

**Comment by CicadaCinema** [comment]
Signed-off-by: Arseny Uskov <arsenyuskov@gmail.com>

**Comment by CicadaCinema** [comment]
I've addressed the comments in https://github.com/vector-im/element-android/pull/4391 (apart from proactively asking the user to turn this option on if their homeserver has no ice server config) and moved this to a new branch - hence the new PR.

**Comment by CicadaCinema** [comment]
@BillCarsonFr @bmarty

File: vector/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -548,6 +548,8 @@
     <string name="settings_call_category">Calls</string>
     <string name="settings_call_show_confirmation_dialog_title">Prevent accidental call</string>
     <string name="settings_call_show_confirmation_dialog_summary">Ask for confirmation before starting a call</string>
+    <string name="settings_call_ringtone_use_default_stun_title">Allow fallback call assist server</string>
+    <string name="settings_call_ringtone_use_default_stun_summary">Will use %s as assist when your homeserver does not offer one (your IP address will be seen by the stun server during a call)</string>
```
Comment (bmarty) [inline vector/src/main/res/values/strings.xml:552]: `ringtone` is a bit strange here, probably coming from a copy/paste :)

**Review by bmarty — COMMENTED** [review]
I think this is OK now, WDYT @BillCarsonFr ?

File: vector/src/main/res/values/strings.xml (hunk 1 above)
Comment (CicadaCinema) [reply vector/src/main/res/values/strings.xml:552]: > `ringtone` is a bit strange here (for the 2 string keys), probably coming from a copy/paste :)

Yes, I copied the string keys from the old riot-android repo:

https://github.com/vector-im/riot-android/blob/a763655c54c4e465d2ae81e3ecd6a5146564771f/vector/src/main/res/values/strings.xml#L349-L350

**Pull request closed without merging**